src/compiler/types.ts

```diff
@@ -4635,55 +4635,47 @@ namespace ts {
 
     export type CompilerOptionsValue = string | number | boolean | (string | number)[] | string[] | MapLike<string[]> | PluginImport[] | ProjectReference[] | null | undefined;
 
-    export interface CompilerOptions {
-        /*@internal*/ all?: boolean;
+    // The initial string negates the work somewhat internally, yes, but it does mean that machines can pull out the rest of the values from the enum  while allowing all values
+    export type AvailableLibOptions = string | "ES5" | "ES6" | "ES2015" | "ES7" | "ES2016" | "ES2017" | "ES2018" | "ESNext" | "DOM" | "DOM.Iterable" | "WebWorker" | "ScriptHost" | "ES2015.Core" | "ES2015.Collection" | "ES2015.Generator" | "ES2015.Iterable" | "ES2015.Promise" | "ES2015.Proxy" | "ES2015.Reflect" | "ES2015.Symbol" | "ES2015.Symbol.WellKnown" | "ES2016.Array.Include" | "ES2017.object" | "ES2017.Intl" | "ES2017.SharedMemory" | "ES2017.String" | "ES2017.TypedArrays" | "ES2018.Intl" | "ES2018.Promise" | "ES2018.RegExp" | "ESNext.AsyncIterable" | "ESNext.Array" | "ESNext.Intl" | "ESNext.Symbol";

```

Comment:
This will reduce to `string`

Reply:
Yeah, thanks! This is why I wrote the comment above - it's parseable out via tools for the site, even if we don't get the safety inside the compiler (where it's a bit less important anyway)